Explorer/Assets/DCL/Notifications/NotificationsController.cs

```diff
@@ -0,0 +1,106 @@
+
+        private static readonly JsonSerializerSettings SERIALIZER_SETTINGS = new () { Converters = new JsonConverter[] { new NotificationJsonDtoConverter() } };
+
+        private readonly CancellationTokenSource cancellationToken;
+        private readonly IWebRequestController webRequestController;
+        private readonly INotificationsBusController notificationsBusController;
+        private readonly IWeb3IdentityCache web3IdentityCache;
+        private readonly CommonArguments commonArgumentsForSetRead;
+        private readonly StringBuilder bodyBuilder = new ();
+        private readonly URLParameter onlyUnreadParameter = new ("onlyUnread", "true");
+        private readonly URLBuilder urlBuilder = new();
+        private CommonArguments commonArguments;
+        private ulong unixTimestamp;
+        private ulong lastPolledTimestamp;
+
+        public NotificationsController(IWebRequestController webRequestController, INotificationsBusController notificationsBusController, IWeb3IdentityCache web3IdentityCache)
+        {
+            this.webRequestController = webRequestController;
+            this.notificationsBusController = notificationsBusController;
+            this.web3IdentityCache = web3IdentityCache;
+
+            cancellationToken = new CancellationTokenSource();
+            lastPolledTimestamp = DateTime.UtcNow.UnixTimeAsMilliseconds();
+            commonArgumentsForSetRead = new CommonArguments(new URLBuilder().AppendDomain(URLDomain.FromString(NOTIFICATION_READ_URL)).Build());
+
+            GetNewNotificationAsync().Forget();
+        }
+
+        private async UniTaskVoid GetNewNotificationAsync()
+        {
+            do
+            {
+                await UniTask.Delay(TimeSpan.FromSeconds(5));
+
+                if(web3IdentityCache.Identity == null)
+                    continue;
+
+                urlBuilder.Clear();
+                urlBuilder.AppendDomain(URLDomain.FromString(NOTIFICATION_URL))
+                          .AppendParameter(onlyUnreadParameter)
+                          .AppendParameter(new URLParameter("from", lastPolledTimestamp.ToString()));
+                commonArguments = new CommonArguments(urlBuilder.Build());
+
+                unixTimestamp = DateTime.UtcNow.UnixTimeAsMilliseconds();
+                lastPolledTimestamp = DateTime.UtcNow.UnixTimeAsMilliseconds();
+                List<INotification> notifications =
+                    await webRequestController.GetAsync(
+                        commonArguments,
+                        new CancellationToken(),
+                        signInfo: WebRequestSignInfo.NewFromRaw(string.Empty, commonArguments.URL, unixTimestamp, "get"),
+                        headersInfo: new WebRequestHeadersInfo().WithSign(string.Empty, unixTimestamp))
+                                              .CreateFromNewtonsoftJsonAsync<List<INotification>>(serializerSettings: SERIALIZER_SETTINGS);
```

Comment:
You should use a pooled list and `OverwriteFromJsonAsync` so you don't recreate it every time

Reply:
If I'm not mistaken OverwriteFromJsonAsync doesn't support custom serializer settings 🤔